examples/bzlmod/MODULE.bazel

```diff
@@ -1,36 +1,128 @@
+# These properties are either essential metadata of the module (such as the name and version),
+# or affect behavior of the current module and its dependents.
 module(
     name = "example_bzlmod",
     version = "0.0.0",
     compatibility_level = 1,
 )
 
+# The following stanza defines the dependency rules_python.
+# For typical setups you set the version.
+# See the releases page for available versions.
+# https://github.com/bazelbuild/rules_python/releases
 bazel_dep(name = "rules_python", version = "0.0.0")
+
+# The following loads rules_python from the file system.
+# For usual setups you should remove this local_path_override block.
 local_path_override(
     module_name = "rules_python",
     path = "../..",
 )
 
+# The following stanza defines the dependency rules_python_gazelle_plugin.
+# For typical setups you set the version.
+# See the releases page for available versions.
+# https://github.com/bazelbuild/rules_python/releases
+bazel_dep(name = "rules_python_gazelle_plugin", version = "0.0.0")
+
+# The following starlark loads the gazelle plugin from the file system.
+# For usual setups you should remove this local_path_override block.
+local_path_override(
+    module_name = "rules_python_gazelle_plugin",
+    path = "../../gazelle",
+)
+
+# The following stanza defines the dependency for gazelle
+# See here https://github.com/bazelbuild/bazel-gazelle/releases/ for the
+# latest version.
+bazel_dep(name = "gazelle", version = "0.30.0", repo_name = "bazel_gazelle")
+
+# The following stanze returns a proxy object representing a module extension;
+# its methods can be invoked to create module extension tags.
 python = use_extension("@rules_python//python:extensions.bzl", "python")
+
+# This name is passed into python.toolchain and it's use_repo statement.
+# We also use the same name for python.host_python_interpreter.
+PYTHON_NAME = "python3"
+
+# This is the name that is used for the host interpreter
+PYTHON_INTERPRETER = PYTHON_NAME + "_host_interpreter"
+
+# We next initialize the python toolchain using the extension.
+# You can set different Python versions in this block.
 python.toolchain(
-    name = "python3_9",
+    # This name is used in the various use_repo statements
+    # below, and in the local extension that is in this
+    # example.
+    name = PYTHON_NAME,
     configure_coverage_tool = True,
     python_version = "3.9",
 )
-use_repo(python, "python3_9")
-use_repo(python, "python3_9_toolchains")
 
+# Import the python repositories generated by the given module extension
+# into the scope of the current module.
+# All of the python3 repositories use the PYTHON_NAME as there prefix.  They
+# are not catenated for ease of reading.
+use_repo(python, PYTHON_NAME)
+use_repo(python, "python3_toolchains")
+
+# This extension is used to look up the host system interpreter.
+# A label that is created by this extension is used for the
+# python_interpreter_target.
+python.host_python_interpreter(
```

Comment:
I don't think we need this separate tag class anymore. The toolchain() call can just create the {name}_host_interpreter repo.

Reply:
Will that give us a unquie name when we have multiple tool chains?

Reply:
Yes, each toolchain() call gets a name, and those names must be unique because eventually they create repos based on that name.

Reply:
@rickeylev addressed